Automattic/woocommerce-payments
# Improve disputed order notice so is more readable and actionable (and not all in bold)

Fixes #7568

#### Changes proposed in this Pull Request

This PR updates the notice displayed in the Woo order screen when the order is disputed. The previous notice had all text in bold.

There are multiple variations of the notice:

- Dispute vs. Inquiry 
- Increasing urgency – > 7 days, < 7 days, < 1 day

The notice is also parameterised to clearly show exact dispute reason and deadline, in a consistent format across all notices.

Here are some screenshots of a cross section of scenarios:

(note, all variations use `'` single quotes for reason now, incorrect in these screenshots, fixed in fd1c04b54f7d8aee5816e49754fbc2838baf42b4)

<img width="1110" alt="Dispute hours" src="https://github.com/Automattic/woocommerce-payments/assets/4167300/859a8f23-a772-4e3e-92cd-583d6ac94f31">
<img width="1085" alt="Dispute normal" src="https://github.com/Automattic/woocommerce-payments/assets/4167300/2dd1a236-b778-47c0-9092-933129f6e8cf">
<img width="1073" alt="Inquiry 5 days" src="https://github.com/Automattic/woocommerce-payments/assets/4167300/4a3b1b6a-950e-4cad-8e58-82233ac8a0e8">
<img width="1063" alt="Inquiry hours" src="https://github.com/Automattic/woocommerce-payments/assets/4167300/24a482ee-f607-4673-b478-9642e37e8435">
<img width="1071" alt="Inquiry normal" src="https://github.com/Automattic/woocommerce-payments/assets/4167300/664ca0e0-e8fd-4eb1-bad3-1dff7ddf60ea">
<img width="1109" alt="Dispute 5 days" src="https://github.com/Automattic/woocommerce-payments/assets/4167300/f05a7114-aa34-4622-8bb4-d897a61e54c5">

#### Implementation notes
The components have been refactored a little to achieve this – the original component had a few conditionals. The goal with this refactoring is to make the code less general, easier to to read, and only support the combinations we need (I found this unclear with the previous combined component).

#### Testing instructions

* Set up orders with various disputes – can use dispute test cards for this.
* View WooCommerce > Orders > [order detail] page for affected orders.
* Ensure the notice appears (note it renders async after a short delay), the info is accurate, and the call to action is clear.

Also compare with old notice to ensure other details are consistent (e.g. button text).

-------------------


**Post merge**
